src/AsyncBridge/AsyncCompatLibExtensions.cs

```diff
@@ -85,5 +86,41 @@ public static ConfiguredTaskAwaitable ConfigureAwait(this Task task, bool contin
 
         return new ConfiguredTaskAwaitable(task, continueOnCapturedContext);
     }
+
+    /// <summary>
+    /// Causes a cancellation token source to cancel after a specified time
+    /// </summary>
+    /// <param name="cancelSource">The cancellation token source to cancel</param>
+    /// <param name="millisecondsDelay">The time in milliseconds to wait before cancellation</param>
+    public static void CancelAfter(this CancellationTokenSource cancelSource, int millisecondsDelay)
```

Comment:
How do you feel about copying the documentation from https://github.com/Microsoft/referencesource/blob/b31308b03e8bd5bf779fb80fda71f31eb959fe0b/mscorlib/system/threading/CancellationTokenSource.cs#L436-L459, since this is a polyfill?

Reading that, I see this:

```cs
/// Subsequent calls to CancelAfter will reset the millisecondsDelay for this  
/// <see cref="T:System.Threading.CancellationTokenSource"/>, if it has not been
/// canceled already.
```

This makes me hesitant to add this extension method unless we can guarantee the same behavior. If called twice, we need to reset the original timer.

Here's what Rackspace.Threading, another polyfill, does (Apache 2.0): https://github.com/tunnelvisionlabs/dotnet-threading/blob/3e99a9d13476a1e8224d81f282f3cedad143c1bc/Rackspace.Threading/CancellationTokenSourceExtensions.cs#L64